components/script/dom/htmlmediaelement.rs

```diff
@@ -710,16 +714,25 @@ impl HTMLMediaElement {
             Some(url) => url.clone(),
             None => self.blob_url.borrow().as_ref().unwrap().clone(),
         };
+        let element = self.upcast::<Element>();
+        let cors_setting = cors_setting_for_element(element);
         let request = RequestInit {
             url: url.clone(),
             headers,
             destination,
-            credentials_mode: CredentialsMode::Include,
             use_url_credentials: true,
             origin: document.origin().immutable().clone(),
             pipeline_id: Some(self.global().pipeline_id()),
             referrer_url: Some(document.url()),
             referrer_policy: document.get_referrer_policy(),
+            mode: match cors_setting {
```

Comment:
This is following the [potential CORS request algorithm](https://html.spec.whatwg.org/multipage/urls-and-fetching.html#create-a-potential-cors-request), correct? If so, I'd suggest adding a comment linking to the spec somewhere above here.